rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/topic_cache.hpp

```diff
@@ -39,31 +41,48 @@ typedef eprosima::fastrtps::rtps::GUID_t GUID_t;
 class TopicCache
 {
 private:
-  typedef std::map<GUID_t,
-      std::unordered_map<std::string, std::vector<std::string>>> ParticipantTopicMap;
   typedef std::unordered_map<std::string, std::vector<std::string>> TopicToTypes;
+  typedef std::map<GUID_t, TopicToTypes> ParticipantTopicMap;
+  typedef std::vector<std::tuple<GUID_t, std::string, rmw_qos_profile_t>> TopicData;
+  typedef std::unordered_map<std::string, TopicData> TopicNameToTopicData;
```

Comment:
In C++, you can use [type aliases](https://en.cppreference.com/w/cpp/language/type_alias)
```c++
using TopicToTypes = std::unordered_map<std::string, std::vector<std::string>>;
using ParticipantTopicMap = std::map<GUID_t, TopictoTypes>;
using TopicData = std::vector<std::tuple<GUID_t, std::string, rmw_qos_profile_t>>;
using TopicNameToTopicData = std::unordered_map<std::string, TopicData>;
```